### ISSUE TITLE
Allow the developer to set the initially selected tab using `selected` attribute on `vaadin-tab`

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-tabs#51]
To make it easier to have a certain tab pre-selected.

```html
<vaadin-tabs>
  <vaadin-tab>Tab one</vaadin-tab>
  <vaadin-tab selected>Tab two</vaadin-tab>
</vaadin-tabs>
```

**Comment by manolo** [comment]
he can do with `<vaadin-tabs selected=2>` which is the same api than paper and it's easy to bound. Or do you suggest to have both ways and maintain two api's

**Comment by jouni** [comment]
Yeah, I was asking if we could/should support two APIs. This suggestion came from @stramel, actually.

I do see there could be potential issues having two APIs doing basically the same thing. So there are at least these edge cases, where it’s not obvious what should happen:

```html
<vaadin-tabs selected="0">
  <vaadin-tab>Tab one</vaadin-tab>
  <vaadin-tab selected>Tab two</vaadin-tab>
</vaadin-tabs>
```

```html
<vaadin-tabs>
  <vaadin-tab selected>Tab one</vaadin-tab>
  <vaadin-tab selected>Tab two</vaadin-tab>
</vaadin-tabs>
```

**Comment by stramel** [comment]
I took a deeper look at the code. Seems like you guys implement the same public API as `paper-tabs` which would allow for `attr-for-selected`. I think that was what I was really looking for. 

```html
<vaadin-tabs selected="foo" attr-for-selected="name">
  <vaadin-tab name="baz">Tab Baz</vaadin-tab>
  <vaadin-tab name="bar">Tab Bar</vaadin-tab>
  <vaadin-tab name="foo">Tab Foo</vaadin-tab>
</vaadin-tabs>
```

I also noticed you guys are applying the `selected` attribute which again aligns with what `paper-tabs` does.

I do however think it is intriguing to be able to set the initially selected tab with a `selected` attribute.

**Comment by manolo** [comment]
There are two features here, I might agree with the attr-for-selected, but should be discussed/tracked in a different thread.

I'm still not convinced about the selected attr in items, which makes a complicated API. We need to document a lot of edge cases that might confuse the user. For instance what would expect the user in these cases?

```
<vaadin-tabs selected="2">
  <vaadin-tab selected>Tab 0</vaadin-tab>
  <vaadin-tab selected>Tab 1</vaadin-tab>
  <vaadin-tab>Tab 2</vaadin-tab>
</vaadin-tabs>
```

```
<vaadin-tabs selected="{{selected}}">
  <vaadin-tab selected>Tab 0</vaadin-tab>
  <vaadin-tab selected$=[[_isFooSelected()]]>Foo</vaadin-tab>
  <vaadin-tab>Tab 2</vaadin-tab>
</vaadin-tabs>
```

For sure that each reader could have a different solution, hence we need to clearly explain what will happen when the developer mixes different ways of selecting/binding. That is imho a anti-pattern of fight for simplicity ;-)
